### PR TITLE
Fix: supporting gpt-oss HF eagle

### PR DESCRIPTION
**Pull request by h-guo18**
## What does this PR do?

**Type of change:** Bug fix 

**Overview:** 
This PR contains two minor fix to support gpt-oss eagle training:

- Added `head_dim` in default eagle config to prevent Llama inferring the head_dim by `hidden_size/num_heads`. This leads to wrong head dim for models like GPT-oss, where `hidden_size != num_heads * head_dim`.
- Refactored eagle checkpoint export to avoid passing speculative decoding model to `_export_hf_checkpoint`, which triggers error for offline training checkpoints.

Other changes:
- Updated readme for data format instructions
- update redundant `wandb.init()` in eagle_utils.py;

## Usage

Not changed.

## Testing
Tested with gpt-oss-120b with offline training, export, and tested checkpoint on spec-bench.

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes/No 
- **Did you write any new necessary tests?**: Yes/No
- **Did you add or update any necessary documentation?**: Yes/No
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes/No 

## Summary by CodeRabbit

- New Features
  - Added a dedicated speculative-decoding export path for eagle-mode checkpoints and helpers to export speculative checkpoints.

- Documentation
  - Added Data Preparation and Other Datasets guidance, JSONL examples, data synthesis notes, and clarified evaluation: in-framework only for online training.
  - Updated example commands to use input_conversations/daring-anteater.jsonl.

- Refactor
  - Removed legacy speculative-decoding post-processing; unified export control flow.

- Config
  - Default eagle head_dim set to 128.

- Tests
  - Test fixture updated to include head_dim = 64.

- Chores
  - Deferred wandb initialization to runtime (callback construction).

### CODE REVIEW

File: modelopt/torch/speculative/eagle/default_config.py [hunk 1]
```diff
@@ -47,4 +47,5 @@
     "use_mtp_layernorm": False,
     "parallel_draft_step": 1,
     "has_lm_head": False,
+    "head_dim": 64,
```
Comment (yeyu-nvidia) [inline modelopt/torch/speculative/eagle/default_config.py]: default value should be 128 (4096//32)

Comment (h-guo18) [reply modelopt/torch/speculative/eagle/default_config.py]: Thanks for reminding. Shall we set the default config to `hidden_size=8192, num_head=32,head_dim=64`? It seems most existing ckpts have head_dim 64:
https://huggingface.co/yuhuili/EAGLE-LLaMA3-Instruct-70B/blob/main/config.json
https://huggingface.co/nvidia/gpt-oss-120b-Eagle3/blob/main/config.json

Comment (yeyu-nvidia) [reply modelopt/torch/speculative/eagle/default_config.py:50]: Okay. Then lets change the entire default_config to llama3.3 70B for consistency

Comment (yeyu-nvidia) [reply modelopt/torch/speculative/eagle/default_config.py]: https://huggingface.co/yuhuili/EAGLE-LLaMA3-Instruct-70B/blob/main/config.json has head_dim=8192/64=128
I dont think we need to change the default config as it is just llama3 8B.

**Review by yeyu-nvidia — APPROVED** [review]

File: examples/speculative_decoding/README.md [hunk 2]
```diff
@@ -115,7 +115,7 @@ After training draft model, we can evaluate the saved modelopt checkpoint on MT-
 python ar_validate.py --model_path $OUTPUT_DIR
 ```
 
-Alternatively, we can export the checkpoint and run evaluation on serving frameworks. See sections below.
+**Note**: In-framework evaluation is supported only for online training. For offline training checkpoints, please export the model and evaluate it using serving frameworks.
```
Comment (meenchen) [inline examples/speculative_decoding/README.md:135]: qq, what's the difference between online training and offline training?

Comment (h-guo18) [reply examples/speculative_decoding/README.md:135]: In online training we load both the teacher model and student model into GPU DRAM and perform teacher forward + student forward/backward in a training step. 

In offline training, we inference the teacher model first (with HF or TRTLLM), dump hidden states to disks, then train the draft model only with dumped distillation signals. Then the checkpoint does not contains teacher modules. : )
